### DISCUSSION TITLE
Does cachelib support timed persistent cache data

### DISCUSSION THREAD

**Discussion by xiedeyantu** [facebook/CacheLib#114] [Q&A] [answered]

**Comment by leozzx** [comment, marked as answer]
I don't know what is "timed persistent cache data", but what cachelib supports is a kind of offline snapshot. You need to stop ("shutdown") the cache and persist the whole cache state and data.

**Reply by xiedeyantu** [reply]
Generate snapshots periodically to avoid shutdown failure. All data in NVM is unavailable, which can ensure that the recently generated snapshots can be used

**Reply by leozzx** [reply]
As cache must be stopped when generating snapshots, we can't do that periodically in the background. User has to make sure there is no cache access and explicitly call shutdown API to persist the cache data.

**Comment by sathyaphoenix** [comment]
Moving to a discussion since this is not an issue to be addressed
